column_text_format/reader.py

```diff
@@ -35,39 +17,46 @@ def findMeta(location):
     return metafile
 
 #Read a ctf metadata file to create a dataframe. It will read nrows rows from each column file. 
-def read(location, columns, numRows):
-    metafile = findMeta(location)
+def read(location, columns, numRows=None):
+    if location.endswith(METASUFFIX+'.'+META_TYPE):
+        metafile = location
+    else:
+        metafile = findMeta(location)
     dataframe = None
     if(metafile != None):
-        file = open(metafile, "r")
-        metadata = json.loads(file.read())
-        for col in columns:
-            #todo if col not in metafile
-            #processes if the columns are selected as int.
-            columnActual = None
-            if isinstance(col, int):
-                columnActual = col
-            else:
-                for columnID, metaColumns in enumerate(metadata['tables'][0]['tableSchema']['columns']):
-                    if(metaColumns['titles']) == col:
-                        columnActual=columnID
-            if(columnActual != None):
-                colLocation = metadata['tables'][0]['tableSchema']['columns'][columnActual]['url']
-                colName = metadata['tables'][0]['tableSchema']['columns'][columnActual]['titles']
-                readColumn = pd.read_csv(colLocation+".txt", nrows=numRows)
-                readColumn.rename(columns={readColumn.columns[0]:colName}, inplace=True)
-                if dataframe is not None:
-                    dataframe = pd.concat([dataframe, readColumn], axis=1)
+        try:
+            file = open(metafile, "r")
+            metadata = json.loads(file.read())
+            for col in columns:
+                #todo if col not in metafile
+                #processes if the columns are selected as int.
+                columnActual = None
+                if isinstance(col, int):
+                    columnActual = col
+                else:
+                    for columnID, metaColumns in enumerate(metadata['tables'][0]['tableSchema']['columns']):
+                        if(metaColumns['titles']) == col:
+                            columnActual=columnID
+                if(columnActual != None):
+                    colLocation = metadata['tables'][0]['tableSchema']['columns'][columnActual]['url']
+                    colName = metadata['tables'][0]['tableSchema']['columns'][columnActual]['titles']
+                    readColumn = pd.read_csv(colLocation+".txt", nrows=numRows)
+                    readColumn.rename(columns={readColumn.columns[0]:colName}, inplace=True)
+                    if dataframe is not None:
+                        dataframe = pd.concat([dataframe, readColumn], axis=1)
+                    else:
+                        dataframe = readColumn
                 else:
-                    dataframe = readColumn
+                    print(col+" was not found within the metadata file")
+        except:
+                print("Unable to open metafile.", file=sys.stderr)
```

Comment:
Use something like `raise ValueError("Unable to open metafile.")`. There's probably a better error than ValueError.